src/Route.php

```diff
@@ -262,4 +257,42 @@ private function validateHttpMethods(array $methods)
 
         return array_map('strtoupper', $methods);
     }
+
+    /**
+     * @param string $path
+     * @param mixed $middleware
+     * @param null|array $methods
+     */
+    private function triggerDeprecationForNonMiddlewareImplementation($path, $middleware, $methods)
+    {
+        $type = $middleware;
+        if (is_string($middleware) && class_exists($middleware)) {
+            $type = $middleware;
+        } elseif (is_string($middleware) && is_callable($middleware)) {
+            $type = 'callable:' . $middleware;
+        } elseif (is_string($middleware) && ! is_callable($middleware)) {
+            $type = 'string:' . $middleware;
+        } elseif (is_callable($middleware) && is_object($middleware)) {
```

Comment:
I think we can have here just `is_object`, doesn't matter if it is callable or not

Reply:
It does; I want to return the object type here, to be more specific. Other callables will report as the string `callable`.

Reply:
Simplfied to `is_object()` only.